src/deploy/functions/discovery/jsexports/parseTriggers.ts

```diff
@@ -69,18 +69,14 @@ function removeInspectOptions(options: string[]): string[] {
 function parseTriggers(
   projectId: string,
   sourceDir: string,
-  configValues: backend.RuntimeConfigValues
+  configValues: backend.RuntimeConfigValues,
+  envs: backend.EnvironmentVariables
 ): Promise<TriggerAnnotation[]> {
   return new Promise((resolve, reject) => {
-    const env = _.cloneDeep(process.env);
+    const env = { ..._.cloneDeep(process.env), ...envs } as NodeJS.ProcessEnv;
```

Comment:
This may be a breaking change, but I actually think we _shouldn't_ ever have been carrying forward `process.env`. It was always weird undefined behavior before and I'd be OK dropping it for now. IMO, we should be populating exactly the env they'll be getting in production.

Reply:
oh, as a nit, you don't need `_.cloneDeep` anymore since you're already cloning with the splat operator and this is just a simple map so there's no recursive cloning to do.